From c35d82bffc9f5fe96d78b3817f20201c57a2248c Mon Sep 17 00:00:00 2001
From: Aloka Dixit <quic_alokad@quicinc.com>
Date: Wed, 22 Sep 2021 17:00:05 -0700
Subject: [PATCH 02/15] mbssid: retrieve driver capabilities

Retrieve driver capabilities for maximum number of interfaces for
MBSSID and maximum allowed profile periodicity for enhanced MBSSID
advertisements.

Signed-off-by: Aloka Dixit <quic_alokad@quicinc.com>
---
 hostapd/main.c       | 3 +++
 src/ap/hostapd.h     | 5 +++++
 src/drivers/driver.h | 5 +++++
 3 files changed, 13 insertions(+)

--- a/hostapd/main.c
+++ b/hostapd/main.c
@@ -253,6 +253,9 @@ static int hostapd_driver_init(struct ho
 				wpa_printf(MSG_ERROR, "set_wowlan failed");
 		}
 		os_free(triggs);
+
+		iface->mbssid_max_interfaces = capa.mbssid_max_interfaces;
+		iface->ema_max_periodicity = capa.ema_max_periodicity;
 	}
 
 	return 0;
--- a/src/ap/hostapd.h
+++ b/src/ap/hostapd.h
@@ -612,6 +612,11 @@ struct hostapd_iface {
 	/* Previous WMM element information */
 	struct hostapd_wmm_ac_params prev_wmm[WMM_AC_NUM];
 
+	/* Maximum number of interfaces supported for MBSSID advertisements */
+	u8 mbssid_max_interfaces;
+	/* Maximum profile periodicity for enhanced MBSSID advertisements */
+	u8 ema_max_periodicity;
+
 	int (*enable_iface_cb)(struct hostapd_iface *iface);
 	int (*disable_iface_cb)(struct hostapd_iface *iface);
 };
--- a/src/drivers/driver.h
+++ b/src/drivers/driver.h
@@ -2185,6 +2185,11 @@ struct wpa_driver_capa {
 
 	/* Maximum number of supported CSA counters */
 	u16 max_csa_counters;
+
+	/* Maximum number of interfaces supported for MBSSID advertisements */
+	u8 mbssid_max_interfaces;
+	/* Maximum profile periodicity for enhanced MBSSID advertisements */
+	u8 ema_max_periodicity;
 };
 
 
